posixlib/src/main/scala/scala/scalanative/posix/pthread.scala

```diff
@@ -17,9 +18,9 @@ import scala.scalanative.posix.sys.types._
 // SUSv2 version is used for compatibility
 // see http://pubs.opengroup.org/onlinepubs/007908799/xsh/threads.html
 
+@link("pthread")
```

Comment:
I think we should leave this explicit `link`. Yes, we do link pthread anyway, since it's needed in GC, however, it might not be actually needed for every GC, eg. none. If we would link pthread conditionally for GC uses and forget to restore this annotation the build would be broken (especially since it's not widely used)  What do you think?

Reply:
@WojciechMazur  Thank you for reading through this PR.

You do a very good job of describing exactly the eason why this PR would
 like to _add_ the explicit `@link`. That line was not there before (understanding
that `@link` may not have been available when that code was written.)

Reply:
Right, I have no idea why I've confused that with deletion! :D Sorry for bothering you with that.